scripts/benchmarks/benches/syclbench.py

```diff
@@ -0,0 +1,379 @@
+# Copyright (C) 2024 Intel Corporation
+# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+import csv
+import io
+from utils.utils import run, git_clone, create_build_path
+from .base import Benchmark
+from .result import Result
+from .options import options
+
+class SyclBench:
+    def __init__(self, directory):
+        self.directory = directory
+        self.built = False
+        self.setup()
+        return
+
+    def setup(self):
+        build_path = os.path.join(self.directory, 'sycl-bench-build')
+        self.bins = build_path
+
+        if self.built or not options.rebuild:
+            return
+
+        create_build_path(build_path, '')
+
+        repo_path = git_clone(self.directory, "sycl-bench-repo", "https://github.com/mateuszpn/sycl-bench.git", "1e6ab2cfd004a72c5336c26945965017e06eab71")
+        print(f"Clone repo to: {repo_path}")
+
+        print(f"Build path: {build_path}")
+
+        configure_command = [
+            "cmake",
+            f"-B {build_path}",
+            f"-S {repo_path}",
+            f"-DCMAKE_BUILD_TYPE=Release",
+            f"-DCMAKE_CXX_COMPILER={options.sycl}/bin/clang++",
+            f"-DCMAKE_C_COMPILER={options.sycl}/bin/clang",
+            f"-DSYCL_IMPL=dpcpp"
+        ]
+
+        print(f"Run {configure_command}")
+        run(configure_command, add_sycl=True)
+        print(f"Run cmake --build {build_path}")
+        run(f"cmake --build {build_path} -j", add_sycl=True)
+
+        self.built = True
+        return
+    
+class SyclBenchmark(Benchmark):
+    def __init__(self, bench, name, test):
+        self.bench = bench
+        self.bench_name = name
+        self.test = test
+        super().__init__(bench.directory)
+
+    def bin_args(self) -> list[str]:
+        return []
+
+    def extra_env_vars(self) -> dict:
+        return {}
+
+    def unit(self):
+        return "ms"
+
+    def setup(self):
+        self.bench.setup()
+        self.benchmark_bin = os.path.join(self.bench.bins, self.bench_name)
+
+    def run(self, env_vars) -> Result:
+        outputfile = f"{self.bench.directory}/{self.test}.csv"
+        command = [
+            f"{self.benchmark_bin}",
+            f"--warmup-run",
+            f"--num-runs=3",
+            f"--output={outputfile}"
+        ]
+        bin_dir = self.bench.bins
+
+        command += self.bin_args()
+        env_vars.update(self.extra_env_vars())
+
+        print(f"Command: {command}")
+        result = self.run_bench(command, env_vars)
+
+        with open(outputfile, 'r') as f:
+            reader = csv.reader(f)
```

Comment:
I'm not sure if I follow. We open an "output" file that contains some benchmark definitions? Shouldn't that be input file then?

Reply:
This is an output file containing results - when put in file, they are structured as csv and easy to analyze. I must analyze the output, whether it is in stdout or file, to return results. Additionally, single benchmark has repetitions, so I have to find median of results